mm2src/mm2_main/tests/mm2_tests/lightning_tests.rs

```diff
@@ -521,56 +629,41 @@ fn test_lightning_taker_swap() {
         block_on(enable_coins_rick_morty_electrum(&mm_node_2))
     );
 
-    // mm_node_1 is maker
-    let set_price = block_on(mm_node_1.rpc(&json!({
-        "userpass": mm_node_1.userpass,
-        "method": "setprice",
-        "base": "RICK",
-        "rel": "tBTC-TEST-lightning",
-        "price": 0.000001,
-        "volume": 0.1
-    })))
-    .unwrap();
-    assert!(set_price.0.is_success(), "!setprice: {}", set_price.1);
-
-    let orderbook = block_on(mm_node_2.rpc(&json!({
-        "userpass": mm_node_2.userpass,
-        "method": "orderbook",
-        "base": "RICK",
-        "rel": "tBTC-TEST-lightning",
-    })))
-    .unwrap();
-    assert!(orderbook.0.is_success(), "!orderbook: {}", orderbook.1);
-
-    block_on(Timer::sleep(1.));
-
-    // mm_node_2 is taker
-    let buy = block_on(mm_node_2.rpc(&json!({
-        "userpass": mm_node_2.userpass,
-        "method": "buy",
-        "base": "RICK",
-        "rel": "tBTC-TEST-lightning",
-        "price": 0.000001,
-        "volume": 0.1
-    })))
-    .unwrap();
-    assert!(buy.0.is_success(), "!buy: {}", buy.1);
-    let buy_json: Json = serde_json::from_str(&buy.1).unwrap();
-    let uuid = buy_json["result"]["uuid"].as_str().unwrap().to_owned();
-
-    // ensure the swaps are started
-    block_on(mm_node_2.wait_for_log(5., |log| {
-        log.contains("Entering the taker_swap_loop RICK/tBTC-TEST-lightning")
-    }))
-    .unwrap();
-    block_on(mm_node_1.wait_for_log(5., |log| {
-        log.contains("Entering the maker_swap_loop RICK/tBTC-TEST-lightning")
-    }))
-    .unwrap();
-
-    block_on(mm_node_1.wait_for_log(900., |log| log.contains(&format!("[swap uuid={}] Finished", uuid)))).unwrap();
+    // -------------------- Test Lightning Taker Swap --------------------
+    let uuids = block_on(start_swaps(
+        &mut mm_node_1,
+        &mut mm_node_2,
+        &[("RICK", "tBTC-TEST-lightning")],
+        0.0005,
+        0.0005,
+        0.1,
+    ));
+    // Todo: use wait_for_swaps_finish_and_check_status instead after fixing lightning swap events
+    block_on(mm_node_1.wait_for_log(900., |log| log.contains(&format!("[swap uuid={}] Finished", uuids[0])))).unwrap();
+    block_on(mm_node_2.wait_for_log(900., |log| log.contains(&format!("[swap uuid={}] Finished", uuids[0])))).unwrap();
```

Comment:
You are right, 15 minutes is too long to wait for the test to fail.
I used the value that is found in other swaps tests, but Lightning swaps in this test should take less than a minute. I think waiting for logs should be longer, if a problem is making the test fail we might need more than a minute of logs for that.
I think 120 seconds / 2 minutes is a good enough value for this, will update the values, it doesn't matter anyways in the end since this test is ignored for other reasons.